plugins/inputs/linux_wireless/linux_wireless.go

```diff
@@ -0,0 +1,200 @@
+package linux_wireless
+
+import (
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"io/ioutil"
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+)
+
+// default file paths
+const (
+	NET_WIRELESS = "/net/wireless"
+	NET_PROC     = "/proc"
+)
+
+// env variable names
+const (
+	ENV_ROOT     = "PROC_ROOT"
+	ENV_WIRELESS = "PROC_NET_WIRELESS"
+)
+
+type Wireless struct {
+	ProcNetWireless string `toml:"proc_net_wireless"`
+	DumpZeros       bool   `toml:"dump_zeros"`
+}
+
+// WirelessDataa struct to hold the tags, headers (measurements) and data
+type WirelessData struct {
+	Headers []string
+	Data    [][]int64
+	Tags    []string
+}
+
+var sampleConfig = `
+  ## file path for proc file. If empty default path will be used:
+  ##    /proc/net/wireless
+  ## This can also be overridden with env variable, see README.
+  proc_net_wireless = "/proc/net/wireless"
+  ## dump metrics with 0 values too
+  dump_zeros       = true
+`
+
+func (ns *Wireless) Description() string {
+	return "Collect wireless interface link quality metrics"
+}
+
+func (ns *Wireless) SampleConfig() string {
+	return sampleConfig
+}
+
+func (ns *Wireless) Gather(acc telegraf.Accumulator) error {
+	// load paths, get from env if config values are empty
+	ns.loadPath()
+	// collect wireless data
+	wireless, err := ioutil.ReadFile(ns.ProcNetWireless)
+	if err != nil {
+		return err
+	}
+	err = ns.gatherWireless(wireless, acc)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (ns *Wireless) gatherWireless(data []byte, acc telegraf.Accumulator) error {
+	wirelessData, err := loadWirelessTable(data, ns.DumpZeros)
+	if err != nil {
+		return err
+	}
+	// go through the WirelessData struct and create maps for
+	// Telegraf to deal with, then addd the data to the
+	// telegraf accumulator
+	for x := 0; x < len(wirelessData.Tags); x++ {
+		entries := map[string]interface{}{}
+		tags := map[string]string{
+			"interface": wirelessData.Tags[x],
+		}
+		for z := 0; z < len(wirelessData.Data[x]); z++ {
+			entries[wirelessData.Headers[z]] = wirelessData.Data[x][z]
+		}
+		acc.AddFields("wireless", entries, tags)
+	}
+	return nil
+}
+
+func loadWirelessTable(table []byte, dumpZeros bool) (WirelessData, error) {
+	wd := WirelessData{}
+	var value int64
+	var err error
+	myLines := strings.Split(string(table), "\n")
+	// split on '|' and trim the spaces
+	h1 := strings.Split(myLines[0], "|")
```

Comment:
Check length of `myLines` to avoid a possible panic here.

Reply:
Also added error return here.